### DISCUSSION TITLE
Massive re-renders with react-navigation on inter-dependent swr hooks

### DISCUSSION THREAD

**Discussion by pke** [vercel/swr#1839] [General]
react-navigation mounts each screen component only once (by default). This poses a major problem when using swr because it causes massive re-renders of hooks that are mounted on each (non visible) screen.

I have many hooks that are dependent on a `useSession` hook and need to update when a property of the session changes. All those hooks re-validate even when the component is not rendered (by react-navigation).

I know its not only an swr problem but do any of you have an idea how to solve this issue?

**Comment by icyJoseph** [comment]
Following from: https://github.com/vercel/swr/discussions/1782#discussioncomment-2271725, and https://github.com/vercel/swr/discussions/1782#discussioncomment-2271990

---

Your last comment does make sense, but I wonder if you have investigated the possibility to use custom focus events. As [document here](https://swr.vercel.app/docs/advanced/react-native#customize-focus-and-reconnect-events).

```js
<SWRConfig
  value={{
    provider: () => new Map(),
    initFocus(callback) {
      /* Register the listener with your state provider */
    },
  }}
>
  <App />
</SWRConfig>
```
